### ISSUE TITLE
Unexpected behavior of getter methods for invalid dates obtained from parsing with patterns

### ISSUE THREAD

**Issue by wmnnd** [moment/moment#2619]
When parsing invalid strings with patterns such as `'M', 'Y', L'`(also when using strict mode), a moment.js object is returned that gives the current date when using getter methods such as `month()` or `year()` instead of an error.
`format()` still returns 'Invalid date' as expected.

````js
var foo  = moment('xyz', 'L', true)
foo.month()
//=> 8
// expected: NaN
foo.year()
// => 2015
// expected: NaN
foo.format('l')
// => 'Invalid date'
````

Without parsing patterns, the following expected behavior is shown:
````js
var bar  = moment('xyz')
bar.month()
//=> NaN
bar.year()
// => NaN
bar.format('l')
// => 'Invalid date'
````

I tested this with moment.js 2.10.6.

**Comment by mj1856** [comment]
Without a pattern, it's falling back to `Date` object, which treats the string as invalid.

With a pattern, it's using moment's parser.  However, the `true` indicates strict mode is enabled, which clearly the string `xyz` does match the `L` pattern (which is `MM/DD/YYYY` in the default US English locale).

**Comment by wmnnd** [comment]
moment.js seems to be aware that the sting is not valid:

````javascript
var foo  = moment('xyz', 'L', true)
foo.isValid()
// => false
````

**Comment by mj1856** [comment]
Interesting!  Thanks.

**Comment by ichernev** [comment]
Fixed in #2624.
